code/modules/mob/living/carbon/alien/humanoid/queen.dm

```diff
@@ -10,9 +10,74 @@
 	layer = LARGE_MOB_LAYER //above most mobs, but below speechbubbles
 	pressure_resistance = 200 //Because big, stompy xenos should not be blown around like paper.
 	butcher_results = list(/obj/item/reagent_containers/food/snacks/meat/slab/xeno = 20, /obj/item/stack/sheet/animalhide/xeno = 3)
+	gps_name = "Hissing Signal"

```

Comment:
To add a gps signal to the queen you just slap
```
AddComponent(/datum/component/gps, "Hissing Signal")
```
Into /mob/living/carbon/alien/humanoid/royal/queen/Initialize()

Reply:
Nice! Thanks.